ps_facetedsearch.php

```diff
@@ -174,31 +179,43 @@ public function install()
             return false;
         }
 
-        Configuration::updateValue('PS_LAYERED_SHOW_QTIES', 1);
-        Configuration::updateValue('PS_LAYERED_FULL_TREE', 1);
-        Configuration::updateValue('PS_LAYERED_FILTER_PRICE_USETAX', 1);
-        Configuration::updateValue('PS_LAYERED_FILTER_CATEGORY_DEPTH', 1);
-        Configuration::updateValue('PS_ATTRIBUTE_ANCHOR_SEPARATOR', '-');
-        Configuration::updateValue('PS_LAYERED_FILTER_PRICE_ROUNDING', 1);
-
-        $this->psLayeredFullTree = 1;
-
-        $this->rebuildLayeredStructure();
-        $this->buildLayeredCategories();
-
-        $productsCount = $this->getDatabase()->getValue('SELECT COUNT(*) FROM `' . _DB_PREFIX_ . 'product`');
+        if ($this->uninstallPrestaShop16Module()) {
+            // ToDo: Does `layered_category` stores critical data?
+            $this->rebuildLayeredStructure();
 
-        if ($productsCount < static::LOCK_TEMPLATE_CREATION) {
-            $this->rebuildLayeredCache();
-        }
-
-        $this->rebuildPriceIndexTable();
-        $this->installIndexableAttributeTable();
-        $this->installProductAttributeTable();
+            // ToDo: Does `layered_price_index` stores critical data ? Probably not with `_index`
+            $this->rebuildPriceIndexTable();
 
-        if ($productsCount < static::LOCK_TOO_MANY_PRODUCTS) {
-            $this->fullPricesIndexProcess();
-            $this->indexAttributes();
+            $this->getDatabase()->execute('TRUNCATE TABLE ' . _DB_PREFIX_ . 'layered_filter CHANGE `filters` `filters` LONGTEXT NULL');
+            $this->getDatabase()->execute('DROP TABLE IF EXISTS ' . _DB_PREFIX_ . 'friendly_url');
+            // ToDo: What cache do we need to (re)generate ?
```

Comment:
Cache generation needed here?

Reply:
We mustn't generate cache while upgrading, it takes too many times (really too many)